test/NuGet.Core.Tests/NuGet.Configuration.Test/EnvironmentSupportTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿using NuGet.Test.Utility;
+using System;
+using System.IO;
+using System.Linq;
+using Xunit;
+
+namespace NuGet.Configuration.Test
+{
+    public class EnvironmentSupportTests
+    {
+        string DefaultNuGetConfigurationWithEnvironmentVariable = @"
+<configuration>
+    <config>
+        <add key='repositoryPath' value='%RP_ENV_VAR%\two' />
+    </config>
+</configuration>";
+
+        [Fact]
+        public void GetValueEvalutatesEnvironmentVariable()
+        {
+            //Arrange
+            var expectedRepositoryPath = @"ONE\two";
+
+            using (var nugetConfigFileFolder = TestFileSystemUtility.CreateRandomTestFolder())
+            {
+                var nugetConfigFile = "NuGetDefaults.config";
+                var nugetConfigFilePath = Path.Combine(nugetConfigFileFolder, nugetConfigFile);
+
+                File.WriteAllText(nugetConfigFilePath, DefaultNuGetConfigurationWithEnvironmentVariable);
+
+                Environment.SetEnvironmentVariable("RP_ENV_VAR", "ONE");
+
+                //Act
+                ISettings settings = new Settings(nugetConfigFileFolder, nugetConfigFile);
+
+                //Assert
+                Assert.Equal(settings.GetValue("config", "repositoryPath"), expectedRepositoryPath);
+            }
+        }
+
+        [Fact]
+        public void GetValuesEvalutesEnvironmentVariable()
+        {
+            //Arrange
+            var expectedRepositoryPath = @"ONE\two";
+
+            using (var nugetConfigFileFolder = TestFileSystemUtility.CreateRandomTestFolder())
+            {
+                var nugetConfigFile = "NuGetDefaults.config";
+                var nugetConfigFilePath = Path.Combine(nugetConfigFileFolder, nugetConfigFile);
+
+                File.WriteAllText(nugetConfigFilePath, DefaultNuGetConfigurationWithEnvironmentVariable);
+
+                Environment.SetEnvironmentVariable("RP_ENV_VAR", "ONE");
+
+                //Act
+                ISettings settings = new Settings(nugetConfigFileFolder, nugetConfigFile);
+                
+                //Assert
+                Assert.Equal(settings.GetSettingValues("config").FirstOrDefault().Value, expectedRepositoryPath);
```

Comment:
Please assign settings.GetSettingValues("config") to a temp variable. 
And then, Assert.Single on the collection.
And then, Assert.Equal on the collection.Single().Value.